archey/output.py

```diff
@@ -65,17 +79,75 @@ def append(self, key, value):
+        It then handles text centering additionally to value and colors replacing.
         """
-        # Let's center the entries according to the logo (handles odd numbers)
-        self._results[0:0] = [''] * ((18 - len(self._results)) // 2)
-        self._results.extend([''] * (18 - len(self._results)))
+        # Iterate through the entries and run their output method to add their content.
+        for entry in self._entries:
+            entry.output(self)
+
+        # Let's copy the logo (so we don't modify the constant!)
+        logo = LOGOS_DICT[self._distribution].copy()
+        logo_width = get_logo_width(logo, len(self._colors_palette))
+
+        # Let's center the entries and the logo (handles odd numbers)
+        height_diff = len(logo) - len(self._results)
+        if height_diff >= 0:
+            self._results[0:0] = [''] * (height_diff // 2)
+            self._results.extend([''] * (len(logo) - len(self._results)))
+        else:
+            colored_empty_line = [str(self._colors_palette[0]) + ' ' * logo_width]
+            logo[0:0] = colored_empty_line * (-height_diff // 2)
+            logo.extend(colored_empty_line * (len(self._results) - len(logo)))
+
+        text_wrapper = TextWrapper(
+            width=(get_terminal_size().columns - logo_width),
+            expand_tabs=False,
+            replace_whitespace=False,
+            drop_whitespace=False,
+            break_on_hyphens=False,
+            max_lines=1,
+            placeholder='...'
+        )
+        placeholder_length = len(text_wrapper.placeholder)
+
+        # Using `TextWrapper`, shortens each entry to remove any line overlapping
+        for i, entry in enumerate(self._results):
+            # Shortens the entry according to the terminal width.
+            # We have to remove any ASCII color, or the result would be skewed.
+            wrapped_entry = text_wrapper.fill(ANSI_ECMA_REGEXP.sub('', entry))
+            placeholder_offset = (
+                placeholder_length if wrapped_entry.endswith(text_wrapper.placeholder) else 0
+            )
+
+            # By using previous positions, re-inserts ASCII colors back in the wrapped string.
+            for color_match in ANSI_ECMA_REGEXP.finditer(entry):
+                match_index = color_match.start()
+                if match_index <= len(wrapped_entry) - placeholder_offset:
+                    wrapped_entry = wrapped_entry[:match_index] + \
+                        color_match.group() + \
+                        wrapped_entry[match_index:]
+
+            # Add a color reset character before the placeholder (if any).
+            # Rationale :
+            # We cannot set `Colors.CLEAR` in the placeholder as it would skew its internals.
+            if placeholder_offset:
+                wrapped_entry = wrapped_entry[:-placeholder_length] + \
+                    str(Colors.CLEAR) + \
+                    wrapped_entry[-placeholder_length:]
+
+            self._results[i] = wrapped_entry
```

Comment:
I feel like we should perhaps stick with this style of line-breaking (as suggested in PEP8 [here](https://www.python.org/dev/peps/pep-0008/#maximum-line-length) and [here](https://www.python.org/dev/peps/pep-0008/#should-a-line-break-before-or-after-a-binary-operator)) since the trailing backslash is a bit fragile (especially if whitespace accidentally gets left after it!) and the operators are pretty far away from what they're operating on.

```suggestion
            # By using previous positions, re-inserts ASCII colors back in the wrapped string.
            for color_match in ANSI_ECMA_REGEXP.finditer(entry):
                match_index = color_match.start()
                if match_index <= len(wrapped_entry) - placeholder_offset:
                    wrapped_entry = (wrapped_entry[:match_index]
                                     + color_match.group()
                                     + wrapped_entry[match_index:])

            # Add a color reset character before the placeholder (if any).
            # Rationale :
            # We cannot set `Colors.CLEAR` in the placeholder as it would skew its internals.
            if placeholder_offset:
                wrapped_entry = (wrapped_entry[:-placeholder_length]
                                 + str(Colors.CLEAR)
                                 + wrapped_entry[-placeholder_length:])

            self._results[i] = wrapped_entry
```

(Though I'm not sure if this will throw a PyLint warning for "wrong" indentation?)